lib/error/ApiAuthRequestError.js

```diff
@@ -2,12 +2,15 @@
 
 var utils  = require('../utils');
 
-function ApiAuthRequestError(message, statusCode){
+function ApiAuthRequestError(errorData){
   Error.call(this);
   Error.captureStackTrace(this, this.constructor);
   this.name='ApiAuthRequestError';
-  this.userMessage = this.message = message;
-  this.statusCode = statusCode || 400;
+  this.userMessage = this.message = errorData.userMessage;
+  this.statusCode = errorData.statusCode || 400;
+  if(errorData.error){
```

Comment:
I'd go with the second one, `this.error = errorData.error`. There isn't really any reason why anyone would explicitly like to type-check for `undefined` in this case.